Add projects' informations for maintainers (#807)
PrestaShop/docs: src/content/1.7/project/maintainers-guide/summary-github-lifecycle.md

## The teams

> On GitHub, there are mainly three teams (employees of PrestaShop's company) working together on the project.

Reviewer: I'm not sure about this, where does the Maintainers team fit in?

Author: I edited this part see https://github.com/PrestaShop/docs/pull/807/commits/7875579b171ffd8d381c7a41bee1aa8565dd75ff

## The teams

On GitHub, there are mainly three teams (employees of PrestaShop's company) working together on the project.

> - Core Product Team: this team takes care of the product vision of the software. It has Product Managers, UX Designers and Wording Managers.

Reviewer 2: ```suggestion
- Core Product Team: this team takes care of the product vision of the software. It has Product Managers, UX Designers, and a Product Content Manager.
```

- Quality Assurance (QA) team: this team takes care of ensuring the quality of the software. It has QA analysts and QA automation engineers.

> - Core Developers Team: this team takes care of the code and act as maintainer for the project.

Reviewer: Technically the Code Developers Team is the same as the Maintainers team.

I'd remove the "employees of PrestaShop's company" note above and replace it with a note stating that at the time of writing, most of the team's members are employees of the PrestaShop Company, but not all of them.

Reviewer: There's also the question of project governance and the ambiguous roles of QA and Product Teams...

Author: > Technically the Core Developers Team is the same as the Maintainers team.

Yes, but I talk here of the teams working for PS company and who work on Github

> There's also the question of project governance and the ambiguous roles of QA and Product Teams

Yes, that's another big question that we are studying

Author: See https://github.com/PrestaShop/docs/pull/807/commits/7875579b171ffd8d381c7a41bee1aa8565dd75ff

#### To be tested

**All the issues with an approved Pull Request that is waiting for validation** by the QA team.

_Who moves issues to this column?_ It is the second maintainer who approves the Pull Request (PRs require at least two approvals), he/she must also add the "waiting for QA" label on the Pull Request.

#### To be merged

**All the issues with a Pull Request that is ready to be merged**, once approved by the QA team.

_Who moves issues to this column?_ It is the QA analyst or the core developer that approves the Pull Request, he/she must also add the "QA approved" label on the Pull Request.

#### Done

All the issues that are now part of the history, great work to all! 

## Pull Requests

When a contributor submits a Pull Request, it goes through multiple stages.

### Is it eligible

Maintainers must first validate that the Pull Request is eligible to review (the template is filled, an issue is linked to the Pull Request, the license headers are correct, the target branch is the right one ...).

If the Pull Request is not eligible, maintainers kindly ask the contributor to fix it and apply the label "Waiting for author" on the Pull Request.

> If the issue associated to the Pull Request is in a project, then the issue should be moved in the "in progress" column.

Reviewer 2: What do you mean by 'in a project'?

Author: In a github project, kanban if you prefer ?

Reviewer 3: Maybe "Github project"

On GitHub, there are mainly three teams working together on the project.

- Core Product Team: this team takes care of the product vision of the software. It has Product Managers, UX Designers, and a Product Content Manager (employees of PrestaShop's company)
- Quality Assurance (QA) team: this team takes care of ensuring the quality of the software. It has QA analysts, QA automation engineers (employees of PrestaShop's company) and issues managers.

> - Core Maintainers Team: this team takes care of the code and act as maintainer for the project (most of them are employees of the PrestaShop Company, but not all of them)

Reviewer 4: Core Maintainers Team or Maintainers Team ?

A maintainer will naturally belong to the Developers team but can also assist and work with the Product Team and QA Team.

## Issues

Currently, there are 4 different types of Issues submitted by users on GitHub.

### Feature request or improvement

When a user submits a feature request or an improvement, it can be either a functional request to add or change a behavior (example: add a new Back office page) or a technical request (example: support PostgreSQL).

Functional requests are labelled "Waiting for PM" while technical requests are labelled "Waiting for dev" by the issue manager (belonging to the QA team) in charge of qualifying the issues.

The Core Product team or the Core Maintainers team in charge will then analyse the request and either reject it or accept it. Rejected requests are closed with "no change required" label, while accepted requests are added to the backlog and labelled as "to be specified" if researches or specifications are needed or "to do". Labels will help triaging the backlog.

### Bug report

> When a user submits a bug report, the issue manager will analyse it and attempt to reproduce it. If it can be reproduced and is confirmed to be an issue, it will be labelled as "to do" and added to the backlog.

Reviewer 3: Who is the issue manager ? 
Maybe you could add link to the dedicated page. 

http://doc.prestashop.com/pages/viewpage.action?spaceKey=PSRD&title=Issue+Manager+Cheat+Sheet#IssueManagerCheatSheet-TheIssueManagerjob

Author: Thank you for your feedback @ldeprestashop

I just updated the first part with the role of the issue manager which belong the QA team
I didn't link the documentation because it's only accessible for PS employees for now, we will need to move it soon so that it will be accessible to everyone :)

Reviewer 5: We could keep somewhere a list of "who is who" but I dont think this should be in a ... developer documentation 😅  . Even this page is not really at the right place. We are not writing documentation here 😛  .

### Support request

Sometimes support requests are submitted on GitHub, ranging from questions about the software to "please help me to do X in my shop" requests. GitHub is for the software development so usually these requests are redirect to other channels using the [Support template](https://github.com/PrestaShop/PrestaShop/issues/new?template=3_support_question.md).

### Other

There are some issues which do not fit in the previous categories, such as [releases Issues](https://github.com/PrestaShop/PrestaShop/issues/20804) to follow the progress of a release, or [epic issues](https://github.com/PrestaShop/PrestaShop/issues/9723) used to group together all requests concerning the same subject.

## Projects

Available in the ‘[Projects](https://github.com/PrestaShop/PrestaShop/projects)’ tab of the PrestaShop’s GitHub repository, versions kanbans define the scope of a minor version in terms of what the Core Maintainers and Core Product teams agree must be included before that release is out.

> Both the product and the maintainers teams use it to prepare and plan the next developments and, since we deal with an open source project, the purpose of having those kanbans public is to help the community and the external mainteners to know what is at stake.

Author: Yes, and everything we do is public because we deal with an open source project :)

> Both the product and the maintainers teams use it to prepare and plan the next developments and, since we deal with an open source project, the purpose of having those kanbans public is to help the community and the external mainteners to know what is at stake.

Reviewer 5: I agree with @eternoendless that the sentence is a bit weird 😅  . I suggest
```suggestion
Both the product and the maintainers teams use it to prepare and plan the next developments. As PrestaShop is an open source project, those kanbans are public.
```

### Definition of states

> A pull request should always be linked to the issue it fixes. 

Reviewer: I have a feeling that this statement and the one after are unrelated

On GitHub, there are mainly three teams working together on the project.

> - Core Product Team: this team takes care of the product vision of the software. It has Product Managers, UX Designers, and a Product Content Manager (employees of the PrestaShop Company)

Reviewer 5: Is it important to say who is an employee of the company and who isn't ? 🤔

Author: I think for clarity

The work of analyzing and reproducing all submitted bug reports is a very complex one, because there might be a very diverse range of reports. Moreover quite a high number of them are actually not related to the software but to how the shop is being used: the server configuration, the shop configuration, the installed modules and the chosen theme might introduce an unexpected behavior that the user mistakenly blames on a software defect.

This is why so many issues cannot be reproduced by issue managers, but figuring that out might require a long time, multiple attempts, and lots of questions.

If the issue manager is unable to reproduced the problem after multiple attempts, or if the user does not answer after 30 days, the issue is labelled as "Can't reproduce" and closed.

#### Regressions

If the bug report is confirmed, one of the key elements to evaluate is whether it is a regression. A regression is a bug that cannot be observed in the previous PrestaShop version, it means the software quality level has decreased instead of increasing. Regressions are usually milestoned to be fixed in the next patch version.

### Support request

> Sometimes support requests are submitted on GitHub, ranging from questions about the software to "please help me to do X in my shop". GitHub issues are not a Q&A forum, so these requests are usually redirected to other channels using the [Support template](https://github.com/PrestaShop/PrestaShop/issues/new?template=3_support_question.md) and closed.

Reviewer 5: ```suggestion
Sometimes support requests are submitted on GitHub, ranging from questions about the software to "please help me to do X in my shop". GitHub issues are not intended for such requests, so these requests are usually redirected to other channels using the [Support template](https://github.com/PrestaShop/PrestaShop/issues/new?template=3_support_question.md) and closed.
```

### Other

Some issues do not fit in the previous categories, such as [Releases Issues](https://github.com/PrestaShop/PrestaShop/labels/Release) which are used to follow the progress of a release, or [Epic Issues](https://github.com/PrestaShop/PrestaShop/labels/Epic) that group together all requests belonging to the same subject.

## Projects

Available in the "[Projects](https://github.com/PrestaShop/PrestaShop/projects)" tab of PrestaShop’s GitHub repository, versions kanban boards define the scope of a minor version in terms of what the Core Maintainers and Core Product teams agree must be included before that release is out.
Both the product and the maintainers teams use it to prepare and plan the next developments and, since we deal with an open source project, the purpose of having those kanbans public is to help the community and the external mainteners to know what is at stake.

### Patches versions

> There should only have regressions issues in the kanban of a patch version. For example, 1.7.7.1 and 1.7.7.2 (and so on...) kanbans only includes the regressions of the 1.7.7.0 version.

Reviewer 5: Maybe mention that sometimes we also include critical or impactful fixes when the timing is important?

For example I remember we added the fix for https://github.com/PrestaShop/PrestaShop/issues/22430 because it was considered an important issue.

Also we dont mention vulnerabilities fixes.

Author: I added security & critical issues

### Minor versions

> There should only have "must-have" issues in the kanban of a minor/major version and it always have an owner, be it a product manager or a developer.

Reviewer 5: ```suggestion
There should only have "must-have" issues in the kanban of a minor/major version and it always has an owner, be it a product manager or a developer.
```
The term "owner" might need to be explained.

> Only "must-have" issues are included in the kanban, as it must reflect the progress of the version. For example, an issue reports a must-have bug prioritized for the 1.7.8.0, the issue must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Reviewer 5: ```suggestion
Only "must-have" issues are included in the kanban, as it must reflect the progress of the version. For example, if a GitHub issue reports a must-have bug milestoned for version 1.7.8.0, it must appear in the 1.7.8 kanban while the linked pull request is milestoned for 1.7.8.0.
```

### Nice to have

"Nice to have" issues are part of another [kanban](https://github.com/PrestaShop/PrestaShop/projects/13) and are not intended for a particular version.

## Definition of states

Community contributions that fix a feature request or an improvement, must be approved by the Core product team if they bring a change in the functional behavior and by the Core developer team if they bring a change in the technical behavior. Ideally before being reviewed, at least before being merged. Here is why:

> It is a bug? The specifications must be completed only when it's incomplete or when there's a behavior change.

Reviewer 5: ```suggestion
It is a bug? The specifications must be completed only when it's not complete or when there's a behavior change.
```
I dont fully understand this sentence. I get the vague idea though.

"Incomplete" is French 😊

> It is a feature request or an improvement? It must be validated, prioritized, and specified. Specification can be written by a contributor or by the Core Product team. The Core Product team has the final word in whether the specification is valid or not.

Reviewer 5: ```suggestion
It is a feature request or an improvement? It must be validated, prioritized, and specified. Specification can be written by a contributor or by the Core Product team. The Core Product team decides whether the specification is valid or not.
```
A little suggestion to make the sentence a bit more friendly.

Your version sounds a bit like "I am the boss!" 😅

A pull request that shows no activity for more than three months must be closed. If it is fixing a major bug or any topwatcher issue, it will be back in the to do column again to be completed by a developer.

### Definition of ready

An important notion to keep in mind is the definition of ready. Several elements must be prepared upstream to get the issue ready to be worked on, otherwise it will not be included in the backlog.

An issue is ready when it contains:
- an explicit title and description
- validated & available specifications
- validated & available mockups
- validated & available wording
- validated & available acceptance tests

#### Not ready

**All the must-have issues that should be included in this version** but that are not yet ready - need to be assigned to an owner, be it a Product Manager or a Lead Developer.

_Who adds issues to this column?_ It is the core product team members once they choosed to prioritize the issue as a "must-have", but it could also be the lead developer with the approval of a Product Manager.

#### Backlog

> **All issues that are considered _ready_** by the Product Managers and the lead dev. Issues in this column will be discussed with the core maintainers.

Reviewer 5: ```suggestion
**All issues that are considered _ready_** by the Product Managers and the Lead Developer. Issues in this column will be discussed with the core maintainers.
```
Lead Developers ? Is it just @eternoendless or do @PierreRambaud and me qualify ? 😄

There are some Issues which do not fit in the previous categories, such as [releases Issues](https://github.com/PrestaShop/PrestaShop/issues/20804). They serve a specific purpose.

## Projects

Available in the ‘[Projects](https://github.com/PrestaShop/PrestaShop/projects)’ tab of the PrestaShop’s GitHub repository, versions kanbans organize the Core team’s work. Both the product and the tech teams use it to prepare and plan the next developments and, since we deal with an open source project, the purpose of having those kanbans public is to help the community and the external mainteners to know what is at stake.

### Definition of states

> A ticket can either be an issue or a pull request. Most of the time, pull requests fix issues so it is most likely that you run into issues while browsing the kanban. A ticket should always have an owner, be it a PM or a developer.

Reviewer 6: So both PR and issues are accepted in the kanban? I thought the purpose of this update was to avoid having PR in the kanban?

If exceptions are authorized we should define when a PR is acceptable, and maybe explain that an issue is always preferred when possible? and mandatory when it exists (only PR without issues are accepted?)

Author: You are right, I just edited this part here: https://github.com/PrestaShop/docs/pull/807/commits/03e8c670ae1de578f0c042976db69f746a31d9d4

> Community contributions are no longer part of the kanban. Only tickets developed or fixed by the Core team take part in the kanban, as it should reflect the Core team’s work. For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Reviewer 6: ```suggestion
Community contributions are no longer part of the kanban. Only tickets developed or fixed by the Core team are included in the kanban, as it should reflect the Core team’s work. For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.
```

> Community contributions are no longer part of the kanban and they must be approved by the product team before being merged!
> Only ticket developed or fixed by the core team take part in the kanban, that should reflect the core team’s work.
> For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Reviewer 2: ```suggestion
Community contributions are no longer part of the kanban. Only tickets developed or fixed by the Core team take part in the kanban, as it should reflect the Core team’s work. For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Community contributions must be approved by the Core product team. Ideally before being reviewed, at least before being merged. Here is why:
```

It is a bug? It must be reported in the functional specifications.
It is an improvement? It must be validated, prioritized, and specified.

> A pull request that shown no activity for more than three months must be closed. If it is fixing a major bug or any topwatcher ticket, it will be back in the to do column again to be completed by the core team.

Reviewer 6: ```suggestion
A pull request that shows no activity for more than three months must be closed. If it is fixing a major bug or any topwatcher ticket, it will be back in the to do column again to be completed by the core team.
```

> Also, an important notion to keep in mind is the definition of ready. Several elements must be prepared upstream to get the ticket ready to be coded, otherwise it will not take part in the backlog.

Reviewer 6: ```suggestion
Also, an important notion to keep in mind is the definition of ready. Several elements must be prepared upstream to get the ticket ready to be coded, otherwise it will not be included in the backlog.
```

> A ready ticket must contain an explicit title and description referring to:

Reviewer 6: That's the perfect ticket, however I'm not sure we always follow these rules?

Author: No, but we are trying to do our best to follow them :)

Reviewer: "An issue is ready when..."

There are some Issues which do not fit in the previous categories, such as [releases Issues](https://github.com/PrestaShop/PrestaShop/issues/20804). They serve a specific purpose.

## Projects

> Available in the ‘[Projects](https://github.com/PrestaShop/PrestaShop/projects)’ tab of the PrestaShop’s GitHub repository, versions kanbans organize the Core team’s work. Both the product and the tech teams use it to prepare and plan the next developments and, since we deal with an open source project, the purpose of having those kanbans public is to help the community and the external mainteners to know what is at stake.

Reviewer: Kanbans are not meant to organize the Core team's work, else there would only be a single "Core kanban" and we'd only work on what's in there. Kanbans are there first and foremost to define the scope of a minor version in terms of _what the Core and Product teams agree **must be included**_ before that release is out. This explains why we don't add community "nice to haves" unless we mean to see those through ourselves.

Author: So, we should keep in Kanban only the must-have of the version, no matter if they are done by a developer working for PrestaShop Corp or by a community member.
In this case how do we manage "nice to have" ? In another Kanban ?

Reviewer: > In this case how do we manage "nice to have" ? In another Kanban ?

Could be. The deeper question is what exactly do we intend to track on the Kanban. One option would be to set up a kanban for each individual version, down to the patch version. In that case, any issue added to a given kanban would _de facto_ become a "Must have" for that specific version. "Nice to haves" would cease to exist in terms that everything is a must have until it's either done or removed from the kanban. When the kanban is "all done", then the version is ready for release and the kanban can be closed.

Author: I edited this part, see https://github.com/PrestaShop/docs/pull/807/commits/9ea846234d916f584ce10b38a4fa8341910a6bf9

### Definition of states

A ticket can either be an issue or a pull request. A pull request should always be linked to the issue it fixes. 

> That's said, there should only have issues in the kanban and it always have an owner, be it a PM or a developer.

Reviewer: Why speak of tickets? I think it's confusing because the concept doesn't exist in Github and we don't add PRs to the kanbans, only **issues**.

Author: You are right, I replaced all "ticket" mentions by "issue"

> Community contributions are no longer part of the kanban. Only tickets developed or fixed by the Core team are included in the kanban, as it should reflect the Core team’s work. For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Reviewer 4: Is there a definition for Core Team ? Are there maintainers ?

Author: Yes, in the first part of the guide:

> Developers team a.k.a Core Team: this team takes care of the code and act as maintainer for the project.

Reviewer 4: External (not in PrestaShop Corp) maintainer ?

Author: No, we are talking of the internal maintainers here (employees)

Reviewer 4: Can you explicit that ?

Author: Yes, I just add this in the first part of the guide: https://github.com/PrestaShop/docs/pull/807/commits/103aa01248de15cb48f0b0c73bcf6f4feb944eed

Reviewer 4: :heart_eyes:

> Community contributions are no longer part of the kanban. Only tickets developed or fixed by the Core team are included in the kanban, as it should reflect the Core team’s work. For example, an issue reports a bug prioritized for the 1.7.8.0, the ticket must appear in the 1.7.8 kanban while the linked pull request must have the 1.7.8.0 milestone.

Reviewer: I disagree, kanbans reflect what we consider is a priority for a version, regardless of who does it. But we commit to doing it ourselves if no one else does it.

Author: I edited this part, see https://github.com/PrestaShop/docs/pull/807/commits/9ea846234d916f584ce10b38a4fa8341910a6bf9

> Community contributions must be approved by the Core product team. Ideally before being reviewed, at least before being merged. Here is why:

Reviewer: This only applies for features which are not in the kanban

Author: ```suggestion
Community contributions that fix a feature request or an improvement, must be approved by the Core product team if they bring a change in the functional behavior and by the Core developer team if they bring a change in the technical behavior. Ideally before being reviewed, at least before being merged. Here is why:
```
Wdyt of this suggestion ?

> It is a bug? It must be reported in the functional specifications.

Reviewer: You man that the functional specifications must be updated to reflect the change? On the contrary, a bug is when the system behaves differently from the specifications. The specifications must be completed only when it's incomplete on when there's a behavior change (in that case we'd be talking about a feature).

Author: I updated this part here: https://github.com/PrestaShop/docs/pull/807/commits/8cc8e740f68653e8f6596fe50ce6b3ffa471a6b6

> It is an improvement? It must be validated, prioritized, and specified.

Reviewer: I don't agree with prioritization if it's contributed. Specification can be also done by the contributor, but the Product team has the final word in whether the specification is valid or not.

Author: You think community contributions shouldn't be prioritized ?
Let's say there is a PR from a contributor fixing a major issue reported by 5 people and another PR from another contributor fixing a minor issue reported by only 1 person.
If we don't prioritize them then the less important one could be merged before the more important one.

> A pull request that shows no activity for more than three months must be closed. If it is fixing a major bug or any topwatcher ticket, it will be back in the to do column again to be completed by the core team.

Reviewer: Can be completed by someone else as well.

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/3bfda264fab93309bddda31671c4d0a074ed684c

> Also, an important notion to keep in mind is the definition of ready. Several elements must be prepared upstream to get the ticket ready to be coded, otherwise it will not be included in the backlog.

Reviewer: ```suggestion
Also, an important notion to keep in mind is the definition of ready. Several elements must be prepared upstream to get the issue ready to be worked on, otherwise it will not be included in the backlog.
```

> A ready ticket must contain an explicit title and description referring to:

Reviewer: "An issue is ready when..."

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/8b1f140b4227d52aa76f58d600e4d00f0d865a18

- validated & available specifications
- validated & available mockups
- validated & available wording
- available acceptance tests

#### Not ready

> **All the prioritized tickets to include in this version** but that need to be prepared by the core team - assigned to a PM or the lead dev as owner.

Reviewer: The term "ticket" should be replaced with "issue"
```suggestion
**All the issues that should be included in this version** but that are not yet ready - need to be assigned to an owner, be it a Project Manager or a Lead Developer.
```

> _Who adds tickets to this column?_ Mostly the PM once they approved the ticket, but it could also be the developers.

Reviewer: ```suggestion
_Who adds issues to this column?_ Mostly Product team members once they choose to prioritize the issue, but Maintainers can also do it.
```

Author: By maintainers you mean only internal (to PrestaShop Corp) or also external maintainers ?

Author: See https://github.com/PrestaShop/docs/pull/807/commits/e5eaf69f79e242bd4d5c4b9eec4d30d664a9213a

#### Backlog

> **All the ready tickets reviewed during the grooming** by the lead PM and the lead dev and discussed/assigned to a developer during the sprint planning.

Reviewer: This is tricky because Grooming and Planning meetings are private. Better leave them out.
```suggestion
**All issues that are considered _ready_** by the lead PM and the lead dev. Issues in this column will be discussed to Core developers.
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/a24cd0d9c79059a6454232bbe0747d265d0af0d0

> _Who adds tickets to this column?_ It is the lead PM with the approval of the lead dev.

Reviewer: ```suggestion
_Who adds tickets to this column?_ It is the lead PM with the approval of lead developers.
```

#### To do

> **All the ready tickets to be developed**, presented by the owners during the sprint planning.

Reviewer: ```suggestion
**All the issues that are considered _ready_ and validated by the developers**, optionally assigned to one.
```

> _Who adds tickets to this column?_ It is the lead PM with the approval of the lead dev. 

Reviewer: ```suggestion
_Who adds issues to this column?_ It is the lead PM with the approval of Core developers. 
```

#### Blocked / Need specs

> **All the tickets that lack information and need to be respecified** or that are waiting for some previous ticket to be merged.

Reviewer: ```suggestion
**All the issues which have been started but lack information and need to be respecified** or that are blocked by some other task.
```

> _Who adds tickets to this column?_ It is the lead PM, with the approval of the lead dev, after the PM/dev assigned to the ticket has raised the blocking element. Add a comment to precise what is expected.

Reviewer: ```suggestion
_Who adds issues to this column?_ It is the issue owner, after the PM/dev assigned to the issue has explained the blocking element. A comment must be added to describe what is expected.
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/4733d9d60946f41248fda78fc6cf982311d919f7

#### In progress

> **All the tickets that are being coded**, assigned to a developer.

Reviewer: ```suggestion
**All the issues that are being actively worked on**, assigned to developer who's claimed it.
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/e3d8bbdf7fc5fc6dbdadeb2d1d1d523b7a644401

> _Who adds tickets to this column?_ It is the assigned developer, once he/she starts working on it.

Reviewer: ```suggestion
_Who adds issues to this column?_ It is the assigned developer, once he/she starts working on it.
```

#### To be reviewed

> **All the tickets whose code needs to be reviewed** - at least two approvals from core-developers are required.

Reviewer: ```suggestion
**All the issues where a Pull Request has been made and is waiting for code review**.
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/b63ca83e9e88acc00260ff5ae119105d4f031784

> _Who adds tickets to this column?_ It is the developer assigned to the ticket.

Reviewer: ```suggestion
_Who adds issues to this column?_ It is the developer assigned to the issue.
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/481f2b39a63e81daa0dbd983252026342d204201

#### To be tested

> **All the tickets approved that need to be tested** by the QA team.

Reviewer: ```suggestion
**All the issues with an approved Pull Request that is waiting for validation** by the QA team.
```

> _Who adds tickets to this column?_ It is the second developer who reviews and approves the ticket, he must also add the 'waiting for QA' label.

Reviewer: ```suggestion
_Who moves issues to this column?_ It is the second Maintainer who approves the Pull Request (PRs require at least two approvals).
```

Author: Updated here: https://github.com/PrestaShop/docs/pull/807/commits/9c5151011e9bd8c1101f059f971ab6c4f5a4bcf5

#### To be merged

> **All the tickets that are ready to be merged** because approved by the QA team.

Reviewer: ```suggestion
**All the issues that are ready to be merged**, once they are approved by the QA team.
```

> _Who adds tickets to this column?_ It is the QA team that has just validated the ticket.

Reviewer: ```suggestion
_Who moves tickets to this column?_ It is the QA analyst that approves the Pull Request.
```
